network/p2p/gossip/mempool.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package gossip
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/ids"
+)
+
+var ErrDuplicateTx = errors.New("duplicate tx")
+
+type Mempool[T Gossipable] struct {
+	Txs     map[ids.ID]bool
+	metrics *Metrics
+}
+
+func NewMempool[T Gossipable](
+	metrics *Metrics,
+) (*Mempool[T], error) {
+	return &Mempool[T]{
+		Txs:     make(map[ids.ID]bool),
+		metrics: metrics,
+	}, nil
+}
+
+func (m *Mempool[T]) Add(gossipable T) error {
+	txID := gossipable.GossipID()
+	if m.Has(txID) {
+		m.metrics.ObserveIncomingGossipable(txID, DroppedDuplicate)
+		return fmt.Errorf("%w: %s", ErrDuplicateTx, txID)
+	}
+
+	m.Txs[txID] = true
+	return nil
+}
+
+func (m *Mempool[T]) Remove(gossipables ...T) {
```

Comment:
I think this comment from my previous review wasn't addressed, but we can get rid of the variadic param + only take in an id.